src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala

```diff
@@ -388,6 +391,92 @@ class PodsControllerTest extends UnitTest with ScalatestRouteTest with RouteBeha
         jsonResponse should have(executorResources(cpus = 100.0, mem = 100.0, disk = 10.0))
       }
     }
+
+    "update a simple single-container pod from docker image w/ shell command" in {
+      implicit val podSystem = mock[PodManager]
+      val f = Fixture()
+      val controller = f.controller()
+
+      val deploymentPlan = DeploymentPlan.empty
+      f.podManager.update(any, eq(false)).returns(Future.successful(deploymentPlan))
+
+      val postJson = """
+                       | { "id": "/mypod", "networks": [ { "mode": "host" } ], "containers": [
+                       |   { "name": "webapp",
+                       |     "resources": { "cpus": 0.03, "mem": 64 },
+                       |     "image": { "kind": "DOCKER", "id": "busybox" },
+                       |     "exec": { "command": { "shell": "sleep 1" } } } ] }
+                     """.stripMargin
+      val entity = HttpEntity(postJson).withContentType(ContentTypes.`application/json`)
+      val request = Put("/mypod")
+        .withEntity(entity)
+        .withHeaders(`Remote-Address`(RemoteAddress(InetAddress.getByName("192.168.3.12"))))
+
+      request ~> controller.route ~> check {
+        response.status should be(StatusCodes.OK)
+        response.header[Headers.`Marathon-Deployment-Id`].value.value() should be(deploymentPlan.id)
+      }
+    }
+
+    "do not update the pod if pod name is wrong" in {
+      implicit val podSystem = mock[PodManager]
+      val f = Fixture()
+      val controller = f.controller()
+
+      val podId = PathId("/unknownpod")
+
+      val deploymentPlan = DeploymentPlan.empty
+      f.podManager.update(any, eq(false)).returns(Future.failed(new PodNotFoundException(podId)))
+
+      val postJson = """
```

Comment:
btw. if we evere remove a field or sth. from the input json, it would be super-hard to track down all these string things and alter those, because it won't fail in compile time... just saying... 

Maybe we can at least share those through some method since some of those only differ in pod name ... ?

Reply:
@jeschkies what do you think?

Reply:
Hm, if I understood you correctly the issue is when we drop a field from the API these JSON strings would still include it. I think that is fine for now. This is the v2 API and it really should not change that much anymore.